### ISSUE TITLE
cmd/cue: recursive import incorrectly imports strings with colons

### ISSUE THREAD

**Issue by uhthomas** [cue-lang/cue#2331]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
0.4.3-1
</pre>

### Does this issue reproduce with the latest stable release?

Yes.

### What did you do?

Given the yaml:

```yaml
abc: "please see more info at: https://example.com/"
```

```
❯ cue import -R test.yaml
```

### What did you expect to see?

```cue
abc: "please see more info at: https://example.com/"
```

### What did you see instead?

```cue
import yaml656e63 "encoding/yaml"

abc: yaml656e63.Marshal(_cue_abc)
let _cue_abc = {
	"please see more info at": "https://example.com/"
}
```

**Comment by verdverm** [comment]
This seems expected, based on reading the help string

```
-R, --recursive                recursively parse string values
```

**Comment by uhthomas** [comment]
I see this a lot when importing Kubernetes manifests. CRDs usually have something like:

```yaml
description: "NodeSelector is a selector which must be true for the pod to fit on a node. Selector which must match a node's labels for the pod to be scheduled on that node. More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/"
```

Admittedly, it clearly _is_ valid given `yq` seems happy enough to parse it and even convert to json.

```sh
❯ cat a.yaml | yq
? NodeSelector is a selector which must be true for the pod to fit on a node. Selector which must match a node's labels for the pod to be scheduled on that node. More info
: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
```

```sh
❯ cat a.yaml | yq -j
Flag --tojson has been deprecated, please use -o=json instead
{
  "NodeSelector is a selector which must be true for the pod to fit on a node. Selector which must match a node's labels for the pod to be scheduled on that node. More info": "https://kubernetes.io/docs/concepts/configuration/assign-pod-node/"
}
```

That said, it is pretty weird. I find the recursive flag useful for importing config maps and the like. Not sure what the right thing to do here is.

**Comment by verdverm** [comment]
Yeah, the real world is messy, and it's hard to build a one-size fits all solution. My hunch is that allowing users to provide configuration to provide more granular control would be the most scalable option.

```
#ImportConfig: { ... } // match cli flags, and more

config: [pattern=string]: #ImportConfig
```

Users could then apply import configuration to a specific file, glob, or regex. It would also be helpful to support pattern matching at the value level, so we can say "apply this rule if the value matches this pattern". You'd also have to account for rule precedence and ordering...

**Comment by myitcv** [comment]
@uhthomas you have been incredibly unlucky here. This was fixed two commits after `v0.4.3` in f207b0b9285ad043221ff9f3c2207d04e8c32cc0.

The following test passes as of that commit and therefore with current tip (401cab2dcdb3):

```
exec cue import -R x.yaml
cmp x.cue x.cue.golden

-- x.yaml --
abc: "please see more info at: https://example.com/"
-- x.cue.golden --
abc: "please see more info at: https://example.com/"
```

**Comment by uhthomas** [comment]
Wow, that is unlucky. Thank you for sharing the fix! Is there a release for cue planned soon?

**Comment by myitcv** [comment]
v0.5.0 will be released today, along with v0.6.0-alpha.1
